### DISCUSSION TITLE
Using Spacy to extract pharmaceutical active ingredients from madical notes

### DISCUSSION THREAD

**Discussion by movilla1976** [explosion/spaCy#3367] [Help: Best practices]
* Operating System: Centos 7
* Python Version Used: 2.75
* spaCy Version Used: 2.0
* Environment Information:

I'm starting with Spacy and natural language processing. By the moment I need a very easy task but, to be honest, it is taking too much time. This is the thing:

- I have a list of ~3000 pharmaceutical active ingredients.
- I have a lot of clinical notes from several hospitals.
- I must build a report of the pharmaceutical active ingredients included in the clinical notes.

At the moment, I'm trying to create a new entity "Pharmaceutical Active Ingredient" and train Spacy to learn all of them. But I'm not sure if this is the right way, as what I need to detect is the exact name of the pharmaceutical active ingredients, and maybe the right way could be a match process.

On the other hand, I can't figure out how to load these 3000 pharmaceutical active ingredients to train Spacy to recognise them.

I would really appreciate your help in this issue.

Thanks a lot and best regards,

Javier Movilla
javi.movilla@gmail.com

**Comment by honnibal** [comment]
I think you want the `Matcher` or `PhraseMatcher` classes. See here: https://spacy.io/usage/linguistic-features#section-rule-based-matching
